tools/management/commands/build_bias_data.py

```diff
@@ -0,0 +1,873 @@
+    def process_referenced_assays(data):
+        '''
+        separate tested assays and reference assays
+        '''
+        for j in data.items():
+            assays, reference = Command.return_refenced_assays(j[1]['assay'])
+            j[1]['assay_list'] = assays
+            j[1]['reference_assays_list'] = reference
+            j[1].pop('assay')
+        return data
+
+    _reference_assay_counter = 0
+    _tested_assay_counter = 0
+    @staticmethod
+    def return_refenced_assays(assays):
+        main, reference = list(), list()
+        for assay in assays:
+            # TODO: change to primary_Endogenous
+            if (assay['bias_reference'] == 'Ref. and principal endo.' or
+            assay['bias_reference'] == 'Endogenous' or
+            assay['bias_reference'] == 'Principal endogenous' or
+            assay['bias_reference'] ==  'Ref. and endo.'):
+                if assay['quantitive_activity'] is not None:
+                    reference.append(assay)
+                    Command._reference_assay_counter = Command._reference_assay_counter+1
+            else:
+                main.append(assay)
+                Command._tested_assay_counter =Command._tested_assay_counter +1
+        main = Command.fetch_endogenous_assay(main, reference)
+        return main, reference
+
+    @staticmethod
+    def fetch_endogenous_assay(main, references):
+        result_list = list()
+        for assay in main:
+            temp_reference_list = list()
+            for reference in references:
+                if assay['family'] == reference['family']:
+                    # if assay['signalling_protein']:
+                    if assay['signalling_protein'] == reference['signalling_protein']:
+                        if assay['assay_type'] == reference['assay_type']:
+                            if assay['cell_line'] == reference['cell_line']:
+                                if assay['measured_biological_process'] == reference['measured_biological_process']:
+                                    temp_reference_list.append(reference)
+                    # else:
+                    #     if assay['assay_type'] == reference['assay_type']:
+                    #         if assay['cell_line'] == reference['cell_line']:
+                    #             if assay['measured_biological_process'] == reference['measured_biological_process']:
+                    #                 temp_reference_list.append(reference)
+
+            if len(temp_reference_list)>0:
+                if len(temp_reference_list)>1:
+                    final_end = None
+                    for _reference_assay in temp_reference_list:
+                        if _reference_assay['bias_reference'] == "Principal endogenous" or _reference_assay['bias_reference'] == "Ref. and principal endo.":
+                            assay['endogenous_assay'] = _reference_assay
+                            final_end = _reference_assay
+                            break
```

Comment:
Have you tried running the code with this update? How much does the data change? Do we lose or gain data points compared to the previous iteration?

Reply:
we gained 1 extra experient with 5 assays annotated